dkg/exchanger.go

```diff
@@ -29,6 +34,46 @@ const (
 	sigValidatorRegistration sigType = 103
 )
 
+// sigTypeStore is a shorthand for a map of sigType to map of core.PubKey to slice of core.ParSignedData.
+type sigTypeStore map[sigType]map[core.PubKey][]core.ParSignedData
+
+// dataByPubkey maps a sigType to its map of public key to slice of core.ParSignedData..
+type dataByPubkey struct {
```

Comment:
personal style: I would not introduce more types. I would just put the lock  and a `map[sigType]sigData` in `exchanger` itself. 

func (e exchanger) set(sigdata) {...}
func (e exchanger) get(sigtype) (sigdata, bool) {...}

But that is just me #lessismore